src/java/org/apache/cassandra/io/util/TrackedDataInputPlus.java

```diff
@@ -49,53 +66,61 @@ public void reset(long count)
 
     public boolean readBoolean() throws IOException
     {
+        checkCanRead(1);
```

Comment:
Do you want to use `TypeSizes` instead here?

Reply:
I think it is more clear to use the size number 1/2/4/8 ,which make the code more readable as the code below just add the number

Reply:
Not following. Why would you need to refactor anything? You guys think `1` is more readable than `TypeSizes.BOOL_SIZE` and on top of that you're hardcoding a value?

`checkCanRead(TypeSizes.BOOL_SIZE)` reads good in my eyes :shrug:

Reply:
I agree with your point of view. From the perspective of code readability, TypeSizes.BOOL_SIZE is definitely much better than hard code 1.
What I said above means that if the existing code is already written as 1/4/8, our newly added code also fills in 1/4/8, then it just echoes back and forth, so the readability is not bad.
Of course, if Jacek is willing to change the existing numbers to TypeSizes, I definitely support your point of view.

Reply:
(1)`checkCanRead(TypeSizes.BOOL_SIZE);
        byte b = source.readByte();
        bytesRead += TypeSizes.BOOL_SIZE;
        return b;`
 (2) `checkCanRead(1);
        byte b = source.readByte();
        bytesRead += 1;
        return b;`
(3)`checkCanRead(TypeSizes.BOOL_SIZE);
        byte b = source.readByte();
        bytesRead += 1;
        return b;`
I think the code itself should explain its behavior. 
just from my point of view, this order is (1) > (2) > (3)